src/ray/gcs/tables.h

```diff
@@ -820,22 +820,6 @@ class ErrorTable : private Log<JobID, ErrorTableData> {
     prefix_ = TablePrefix::ERROR_INFO;
   };
 
-  /// Push an error message for the driver of a specific.
-  ///
-  /// TODO(rkn): We need to make sure that the errors are unique because
-  /// duplicate messages currently cause failures (the GCS doesn't allow it). A
-  /// natural way to do this is to have finer-grained time stamps.
-  ///
-  /// \param job_id The ID of the job that generated the error. If the error
-  /// should be pushed to all drivers, then this should be nil.
-  /// \param type The type of the error.
-  /// \param error_message The error message to push.
-  /// \param timestamp The timestamp of the error.
-  /// \return Status.
-  // TODO(qwang): refactor this API to implement broadcast.
```

Comment:
@jovany-wang Just curious what does the broadcast indicate here?

Reply:
I have confirmed to qwang and he doesn't remember.

Reply:
I took a note here for that I prefer using 2 APIs for it:
```
PushErrorToDriver(job_id, xxx);

PushErrorToAll(xxx);  // this is a broadcast to push error to every one.
```

Because the parameter `job_id` indicates the job that generated this error rather than the destination driver, `PushErrorToDriver(Nil(), )`doesn't match the semantic since Nil is not the job generating this error.

Reply:
How about using two APIs below:

```suggestion
AsyncReportJobError(error_data, callback);
AsyncReportRayError(error_data, callback);
```

Reply:
I prefer `PushErrorToDriver(job_id, error_type, message)` and `PushErrorToAllDrivers(error_type, message)`. Current name `AsyncReportError` is confusing, it sounds like reporting an error of raylet to GCS. 
Also, we should avoid exposing `table` in API.

Reply:
Actually, `ReportJobError` means something wrong happened to the `Job` and the error is now reported to GCS. It didn't concern who will subscribe the error, anyone who cares can subscribe the error. 
And there is some other errors, e.g.:  "worker pool large (due to result of using a large number of actors..)"; "node removed (the node has been marked dead)". Those errors means     something happened to ray, maybe not only expose to driver?

Reply:
And I think it's better not list all parameters of the API, since we might change the parameters.

Reply:
`ReportJobError` sounds better than `AsyncReportError`. 
I think listing all parameters is fine, parameters shouldn't change very often.  The main problem with the table parameter is that users need to know too many unnecessary details. E.g., `job_id = nil` means pushing to all drivers, and timestamp can also be automatically generated inside the client.

Reply:
`Worker` might report error to `Raylet` with timestamp inited.

Reply:
So the conclusion is:

Reply:
So is it ok to you:
```
Status AsyncReportJobError(job_id, type, error_message, timestamp, callback);
Status AsyncReportRayError(type, error_message, timestamp, callback);

```